### PR TITLE
Port BaT benchmarks

### PR DESCRIPTION
**Pull request by carneyweb**
Ported the benchmarks from existing MathWorks benchmarks.

### CODE REVIEW

File: framework/test/bench/AnyMapPerfTest.cpp [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+
+class AnyMapPerfTestFixture : public ::benchmark::Fixture
+{
+public:
+  using benchmark::Fixture::SetUp;
+  using benchmark::Fixture::TearDown;
+
+  void SetUp(const ::benchmark::State&)
+  {
+    using namespace cppmicroservices;
```
Comment (jeffdiclemente) [inline framework/test/bench/AnyMapPerfTest.cpp]: the cppmicroservices namespace was already brought in on line 16

File: framework/test/gtest/AnyMapTest.cpp [hunk 2]
```diff
@@ -149,7 +149,7 @@ TEST(AnyMapTest, AnyMap)
   AnyMap uco_anymap(uco);
 
   AnyMap o_anymap1(AnyMap::ORDERED_MAP);
-  o_anymap1 = o_anymap1;
```
Comment (jeffdiclemente) [inline framework/test/gtest/AnyMapTest.cpp]: why is this commented?

File: framework/test/bench/AnyMapPerfTest.cpp (hunk 1 above)
Comment (carneyweb) [reply framework/test/bench/AnyMapPerfTest.cpp]: Done.

File: framework/test/gtest/AnyMapTest.cpp (hunk 2 above)
Comment (carneyweb) [reply framework/test/gtest/AnyMapTest.cpp]: Removed comment out line.

File: framework/src/util/AnyMap.cpp [hunk 3]
```diff
@@ -146,7 +146,7 @@ Any AtCompoundKey(const std::vector<Any>& v,
         ref_any_cast<std::vector<Any>>(h), tail, std::move(defaultval));
     }
   }
-  return defaultval;
+  return std::move(defaultval);
```
Comment (jeffdiclemente) [inline framework/src/util/AnyMap.cpp]: what's the purpose of these changes? They don't seem to have any relation to the benchmark tests.

Comment (carneyweb) [reply framework/src/util/AnyMap.cpp]: Without that change I get the following warning/error:

      local variable 'defaultVal' will be copied despite being returned by name
      [-Werror,-Wreturn-std-move]
